spec/controllers/api/v1/podcasts_controller_spec.rb

```diff
@@ -3,65 +3,147 @@
-        name: "podcast2",
-        url: "www.podcast2.com")
+      name: "podcast2",
+      url: "http://www.podcast2.com")
     }
 
     it "returns a successful response status and a content type of json" do
-        get :show, params: {id: podcast1.id}
+      get :show, params: {id: podcast1.id}
 
-        expect(response.status).to eq 200
-        expect(response.content_type).to eq 'application/json'
+      expect(response.status).to eq 200
+      expect(response.content_type).to eq 'application/json'
     end
 
     it "returns the specified podcast name and podcast url" do
-        get :show, params: {id: podcast1.id}
+      get :show, params: {id: podcast1.id}
+      response_body = JSON.parse(response.body)
+
+      expect(response_body.length).to eq 5
+
+      expect(response_body["name"]).to eq podcast1.name
+      expect(response_body["url"]).to eq podcast1.url
+
+      expect(response_body["name"]).to_not eq podcast2.name
+      expect(response_body["url"]).to_not eq podcast2.url
+    end
+  end
+
+  describe "POST#new" do
+    let!(:new_podcast_hash) { { podcast: { name: "Reply All", url: "https://gimletmedia.com/shows/reply-all" } } }
+
+    it "creates a new Podcast record" do
+      previous_count = Podcast.count
+      post :create, params: new_podcast_hash, format: :json
+      new_count = Podcast.count
+
+      expect(new_count).to eq(previous_count + 1)
+    end
+
+    it "returns the new Podcast as json" do
+      post :create, params: new_podcast_hash, format: :json
+
+      response_body = JSON.parse(response.body)
+
+      expect(response_body["podcast"].length).to eq 5
+      expect(response_body["podcast"]["name"]).to eq "Reply All"
+      expect(response_body["podcast"]["url"]).to eq "https://gimletmedia.com/shows/reply-all"
+    end
+
+    context "when a malformed request is made" do
+      let!(:bad_podcast_hash_1) { { podcast: { url: "https://gimletmedia.com/shows/reply-all" } } }
+      let!(:bad_podcast_hash_2) { { podcast: { name: "Reply All" } } }
+      let!(:bad_podcast_hash_3) { { podcast: { name: "Reply All", url: "gimletmedia.com" } } }
+      let!(:bad_podcast_hash_4) { { podcast: { name: "Reply All", url: "https://gimletmedia.com/shows/reply-all" } } }
+      let!(:bad_podcast_hash_5) { { podcast: { name: "", url: "" } } }
+
+      it "does not create a new podcast if podcast name is not provided" do
+        previous_count = Podcast.count
+        post :create, params: bad_podcast_hash_1, format: :json
+        new_count = Podcast.count
+
+        expect(new_count).to eq previous_count
+      end
+
+      it "does not create a new podcast if podcast url is not provided" do
+        previous_count = Podcast.count
+        post :create, params: bad_podcast_hash_2, format: :json
+        new_count = Podcast.count
+
+        expect(new_count).to eq previous_count
+      end
+
+      it "returns an error if name and url are blank" do
+        post :create, params: bad_podcast_hash_1, format: :json
```

Comment:
this it block does not match the test - `hash_1` includes a URL but the statement is checking for a blank URL and name